pkg/template/controller/metrics.go

```diff
@@ -36,30 +45,39 @@ func (c *TemplateInstanceController) Collect(ch chan<- prometheus.Metric) {
 		return
 	}
 
-	templateInstancesTotal.Reset()
-	templateInstancesActiveStartTime.Reset()
-
-	templateInstancesTotal.WithLabelValues("", "").Set(0)
+	templateInstanceInfo.Reset()
+	templateInstanceStatusCondition.Reset()
+	templateInstanceActiveCreationTime.Reset()
 
 	for _, templateInstance := range templateInstances {
-		waiting := true
+		active := true
 
-		templateInstancesTotal.WithLabelValues("", "").Inc()
+		templateInstanceInfo.WithLabelValues(templateInstance.Namespace, templateInstance.Name).Inc()
 
 		for _, cond := range templateInstance.Status.Conditions {
-			templateInstancesTotal.WithLabelValues(string(cond.Type), string(cond.Status)).Inc()
+			templateInstanceStatusCondition.WithLabelValues(templateInstance.Namespace, templateInstance.Name, string(cond.Type), "true").Set(boolFloat64(cond.Status == kapi.ConditionTrue))
+			templateInstanceStatusCondition.WithLabelValues(templateInstance.Namespace, templateInstance.Name, string(cond.Type), "false").Set(boolFloat64(cond.Status == kapi.ConditionFalse))
+			templateInstanceStatusCondition.WithLabelValues(templateInstance.Namespace, templateInstance.Name, string(cond.Type), "unknown").Set(boolFloat64(cond.Status == kapi.ConditionUnknown))
```

Comment:
again i understand (after talking to @jim-minter) that this follows KSM, but it seems weird to guarantee we are going to report a datapoint for every combination of condition values, for each template instance.  Reporting "0" doesn't seem terribly interesting for a condition=foo if we're also reporting "1" for condition=bar.

Reply:
(and on that same note it seems heavy handed to report N datapoints for every template instance, rather than just a few datapoints that roll up the totals for each condition).  Again, understand this follows the KSM pattern, but want to call @smarterclayton's attention to it.